modules/RTC/TraceablePeerConnection.js

```diff
@@ -1957,23 +1957,21 @@ TraceablePeerConnection.prototype.setAudioTransferActive = function(active) {
  */
 TraceablePeerConnection.prototype.setMaxBitRate = function(localTrack) {
     const mediaType = localTrack.type;
+    const trackId = localTrack.track.id;
+    const videoType = localTrack.videoType;
 
-    // No need to set max bitrates on the streams in the following cases.
-    // 1. When an audio track has been replaced.
-    // 2. When a 'camera' track is replaced in plan-b mode, since its a new sender.
-    // 3. When the config.js option for capping the SS bitrate is not enabled.
-    if ((mediaType === MediaType.AUDIO)
-        || (browser.usesPlanB() && !this.options.capScreenshareBitrate)
-        || (browser.usesPlanB() && localTrack.videoType === 'camera')) {
+    // Change bitrates only when the new track added to the peerconnection
+    // is a desktop track.
+    if (mediaType === MediaType.AUDIO || videoType === VideoType.CAMERA) {
         return;
     }
     if (!this.peerconnection.getSenders) {
         logger.debug('Browser doesn\'t support RTCRtpSender');
 
         return;
     }
-    const videoType = localTrack.videoType;
-    const trackId = localTrack.track.id;
+    const presenterEnabled = localTrack._originalStream
```

Comment:
wouldn't this also be true for any potentiall future effect on the video stream?

Reply:
Yeah that is true .. any potential future effect that we apply on the desktop stream will alter the max bit rate. I couldn't find another way to detect this.